lib/elixir/src/elixir_erl_pass.erl

```diff
@@ -211,31 +211,37 @@ translate({Name, Meta, Args}, S) when is_atom(Name), is_list(Meta), is_list(Args
 
 %% Remote calls
 
-translate({{'.', _, [Left, Right]}, Meta, []}, S)
+translate({{'.', _, [Left, Right]}, Meta, []}, #elixir_erl{context=Context} = S)
```

Comment:
Let's match on `context=guard` in the function head:

```elixir
translate({{'.', _, [Left, Right]}, Meta, []}, #elixir_erl{context=guard = S)
```

And raise an explicit and clear error if there are parens.

Reply:
Well, we can keep the `guard` check inside as well, let's just make sure to raise errors when the left side is an atom or when parens are used.